tests/integration/hsm-integration-test.cpp

```diff
@@ -275,20 +275,60 @@ int main(void)
     HsmEngine hsmEngine(id, modulePath, tokenLabel, pin);
     utility::stringCleanse(pin);
     std::vector<uint8_t> message = utility::fromHex("deadbeef");
+    // Default specs are used in numerous places
+    ECCSpec eccSpec;
+    RSASpec rsaSpec;
+
+    /************** Key generation and loading **************/
+    std::cerr << "0. Testing key generation and loading:\n";
+    std::vector<uint8_t> emptyKeyId{};
+    std::string emptyLabel{};
+    std::vector<uint8_t> keyId_1{0x11};
+    std::vector<uint8_t> keyId_2{0x12};
+    std::string keyLabel_1{"key-label-1"};
+    std::string keyLabel_2{"key-label-2"};
+
+    std::cerr << "Generating a key without specifying the key ID...";
+    try {
+        auto keypair =
+                AsymmetricPrivateKey::generateKeyOnHSM(hsmEngine, eccSpec, keyLabel_1, emptyKeyId);
+    } catch (const MoCOCrWException &e) {
+        std::cerr << std::string(e.what()) + "...";
+    }
+    std::cerr << "Success\n";
+    std::cerr << "Loading a key without specifying the key ID...";
+    try {
+        auto keypair =
+                AsymmetricPrivateKey::readPrivateKeyFromHSM(hsmEngine, keyLabel_1, emptyKeyId);
+    } catch (const MoCOCrWException &e) {
+        std::cerr << std::string(e.what()) + "...";
+    }
+    std::cerr << "Success\n";
+    std::cerr << "Generating a key without specifying the label...";
+    auto key1 = AsymmetricPrivateKey::generateKeyOnHSM(hsmEngine, eccSpec, emptyLabel, keyId_1);
```

Comment:
Error handling and proper reporting? Currently, it just crashes with "uncaught exception" if this fails?!

Reply:
Added everything to try-catch block